Changes to what is built by default in the FreeBSD port.

===================================================================
--- m3/src/m3makefile.orig	Tue Sep 17 16:47:59 1996
+++ m3/src/m3makefile	Sat Jun  6 10:54:04 1998
@@ -83,6 +83,12 @@
 % requires:
 BuildChunk ("m3core")
 
+%--- m3configvars --- library -------------------------------------------------
+% m3configvars is a small library containing various configuration
+% parameters related to this particular installation of Modula-3.
+% requires: m3core
+BuildChunk ("m3configvars")
+
 %--- libm3 --- library -------------------------------------------------
 % libm3 is the library that most others need.
 % requires: m3core
@@ -99,12 +105,12 @@
 %--- m3middle --- library -------------------------------------------------
 % the Modula-3 compiler's IL definition
 % requires: libm3
-% BuildChunk ("m3middle")  %-- built as part of the bootstrap process
+BuildChunk ("m3middle")
 
 %--- m3linker --- library -------------------------------------------------
 % the Modula-3 prelinker
 % requires: m3middle
-% BuildChunk ("m3linker")  %-- built as part of the bootstrap process
+BuildChunk ("m3linker")
 
 %--- m3objfile --- library -------------------------------------------------
 % the Modula-3 object file writers
@@ -114,7 +120,7 @@
 %--- m3front --- library -------------------------------------------------
 % the Modula-3 compiler front-end
 % requires: m3middle
-% BuildChunk ("m3front")  %-- built as part of the bootstrap process
+BuildChunk ("m3front")
 
 %--- m3back --- library -------------------------------------------------
 % the Windows/NT x86 back-end
@@ -129,7 +135,7 @@
 %--- m3 --- utility -------------------------------------------------
 % the Modula-3 compiler (main program)
 % requires: m3linker, m3front, m3middle, libm3
-% BuildChunk ("m3")  %-- built as part of the bootstrap process
+BuildChunk ("m3")
 
 %--- m3loader --- utility -------------------------------------------------
 % an experimental dynamic loader for Windows/NT
@@ -145,7 +151,7 @@
 
 %--- m3build2 --- utility ------------------------------------------------
 % an experimental Modula-3 version of m3build
-% requires: libm3, m3quake
+% requires: m3configvars, libm3, m3quake
 % BuildChunk ("m3build2")
 
 %--- cg-burs --- utility -------------------------------------------------
@@ -207,7 +213,7 @@
 %--- slisp --- library -----------------------------------------------
 % a library containing a small Lisp interpreter
 % requires: libm3
-BuildChunk ("slisp")
+% BuildChunk ("slisp")
 
 %--- tempfiles --- library -----------------------------------------------
 % a library to build tempfiles
@@ -229,32 +235,32 @@
 %--- netobj --- library -------------------------------------------
 % the network objects runtime library, needed by most distributed applications
 % requires: tcp
-BuildChunk ("netobj")
+% BuildChunk ("netobj")
 
 %--- netobjd --- utility -------------------------------------------
 % the network objects daemon, needed by all programs using network objects
 % requires: netobj
-BuildChunk ("netobjd")
+% BuildChunk ("netobjd")
 
 %--- m3tk --- library -------------------------------------------
 % a (large) Modula-3 abstract syntax tree (AST) toolkit, needed by stubgen
 % requires: libm3
-BuildChunk ("m3tk")
+% BuildChunk ("m3tk")
 
 %--- stubgen --- utility -------------------------------------------
 % the network objects stub generator, needed by programs using network objects
 % requires: m3tk
-BuildChunk ("stubgen")
+% BuildChunk ("stubgen")
 
 %--- stable --- library ---------------------------------------------
 % A library providing log-based persistent objects
 % requires: libm3
-BuildChunk ("stable")
+% BuildChunk ("stable")
 
 %--- stablegen --- utility -------------------------------------------
 % a stub generator to make the use of the "stable" library much easier
 % requires: m3tk, stable
-BuildChunk ("stablegen")
+% BuildChunk ("stablegen")
 
 %--- sharedboard --- prototype -------------------------------------------
 % a prototype shared whiteboard
@@ -273,7 +279,7 @@
 % PEX implements a Modula-3 interface to the PEX 3D graphics library.
 %   It is used by all graphical 3D applications
 % requires: X11R4 and a host PEX implementation
-if PLATFORM_SUPPORTS_PEX
+if PLATFORM_SUPPORTS_DECPEX
   BuildChunk ("PEX")
 end
 
@@ -289,21 +295,21 @@
 % motif implements a Modula-3 interface to the X/Motif library.
 % requires: X11R4
 if PLATFORM_SUPPORTS_MOTIF
-  BuildChunk ("motif")
+%  BuildChunk ("motif")
 end
 
 %--- tetris --- game -------------------------------------------------
 % a Modula-3 version of Tetris
 % requires: X11R4
 if PLATFORM_SUPPORTS_X
-  BuildChunk ("tetris")
+%  BuildChunk ("tetris")
 end
 
 %--- columns --- game -------------------------------------------------
 % a Modula-3 version of the PC game, columns
 % requires: X11R4
 if PLATFORM_SUPPORTS_X
-  BuildChunk ("columns")
+%  BuildChunk ("columns")
 end
 
 %--- ui --- library -------------------------------------------------
@@ -311,23 +317,25 @@
 %   by most graphical applications.
 % requires: X11R4 and tcp on POSIX; libm3 on WIN32
 % Note: The WIN32 version is still VERY buggy
-BuildChunk ("ui")
+if PLATFORM_SUPPORTS_X
+  BuildChunk ("ui")
+end
 
 %--- bicycle --- library -------------------------------------------------
 % bicycle is a library of playing card images.
 % requires: ui
-BuildChunk ("bicycle")
+% BuildChunk ("bicycle")
 
 %--- solitaire --- game -------------------------------------------------
 % a Modula-3 version of SeaHaven towers
 % requires: bicycle
-BuildChunk ("solitaire")
+% BuildChunk ("solitaire")
 
 %--- badbricks --- game -------------------------------------------------
 % a Modula-3 game similar to minesweeper, inspired by the crumbling
 %     facade of SRC's building.
 % requires: ui
-BuildChunk ("badbricks")
+% BuildChunk ("badbricks")
 
 %--- m3tools --- library -------------------------------------------------
 % a simple Modula-3 scanner, needed by m3tohtml and m3browser
@@ -336,12 +344,12 @@
 
 %--- m3tohtml --- utility -------------------------------------------------
 % a program to convert batches of Modula-3 source to interconnected HTML
-% requires: libm3, m3tools
+% requires: m3configvars, libm3, m3tools
 BuildChunk ("m3tohtml")
 
 %--- m3browser --- utility -------------------------------------------------
 % an HTTP server that provides WWW browsing of the installed Modula-3 system
-% requires: libm3, m3tools
+% requires: m3configvars, libm3, m3tools
 if equal (OS_TYPE, "POSIX")
   BuildChunk ("m3browser")
 end
@@ -365,17 +373,19 @@
 % a large collection of useful window widgets, needed by many of the
 %   following libraries and programs.
 % requires: ui
-BuildChunk ("vbtkit")
+if PLATFORM_SUPPORTS_X
+  BuildChunk ("vbtkit")
+end
 
 %--- fours --- game -------------------------------------------------
 % a collection of Modula-3 variants of the PC game, tetris.
 % requires: vbtkit
-BuildChunk ("fours")
+% BuildChunk ("fours")
 
 %--- showheap --- utility -------------------------------------------------
 % a program to graphically display in real-time the state of each heap page.
 % requires: vbtkit
-if equal (OS_TYPE, "POSIX")
+if equal (OS_TYPE, "POSIX") and PLATFORM_SUPPORTS_X
   BuildChunk ("showheap")
 end
 
@@ -389,90 +399,102 @@
 %--- replayheap --- utility -------------------------------------------------
 % a program to graphically display the log captured by "recordheap"
 % requires: vbtkit
-if equal (OS_TYPE, "POSIX")
+if equal (OS_TYPE, "POSIX") and PLATFORM_SUPPORTS_X
   BuildChunk ("replayheap")
 end
 
 %--- shownew --- utility -------------------------------------------------
 % a program to graphically display in real-time per-type allocations
 % requires: vbtkit
-if equal (OS_TYPE, "POSIX")
+if equal (OS_TYPE, "POSIX") and PLATFORM_SUPPORTS_X
   BuildChunk ("shownew")
 end
 
 %--- showthread --- utility -------------------------------------------------
 % a program to graphically display in real-time the state of each thread.
 % requires: vbtkit
-if equal (OS_TYPE, "POSIX")
+if equal (OS_TYPE, "POSIX") and PLATFORM_SUPPORTS_X
   BuildChunk ("showthread")
 end
 
 %--- images --- library -------------------------------------------------
 % support for displaying bitmap images
 % requires: ui
-BuildChunk ("images")
+if PLATFORM_SUPPORTS_X
+  BuildChunk ("images")
+end
 
 %--- jvideo --- library -------------------------------------------------
 % a low-level interface to the J-video hardware, needed by videovbt
 % requires: tcp
-BuildChunk ("jvideo")
+if PLATFORM_SUPPORTS_X
+  BuildChunk ("jvideo")
+end
 
 %--- videovbt --- library -------------------------------------------------
 % a window widget that displays live video images
 % requires: jvideo, ui
-BuildChunk ("videovbt")
+if PLATFORM_SUPPORTS_X
+  BuildChunk ("videovbt")
+end
 
 %--- formsvbtpixmaps --- library -------------------------------------------
 % misc. bitmaps, cursors and stuff used by formsvbt
 % requires: libm3, m3bundle
-BuildChunk ("formsvbtpixmaps")
+if PLATFORM_SUPPORTS_X
+  BuildChunk ("formsvbtpixmaps")
+end
 
 %--- formsvbt --- library -------------------------------------------------
 % a high-level language based on S-expressions that makes it easy
 % to assemble VBTs (windows) using the TeX metaphors of boxes and
 % glue.
 % requires: vbtkit, videovbt, formsvbtpixmaps, web
-BuildChunk ("formsvbt")
+if PLATFORM_SUPPORTS_X
+  BuildChunk ("formsvbt")
+end
 
 %--- formsedit --- utility -------------------------------------------------
 % a 1-1/2 view GUI editor for FormsVBT expressions
 % requires: formsvbt
-BuildChunk ("formsedit")
+if PLATFORM_SUPPORTS_X
+  BuildChunk ("formsedit")
+end
 
 %--- fisheye --- demo -------------------------------------------------
 % a demo of "fisheye" views for graph browsing
 % requires: formsvbt
-BuildChunk ("fisheye")
+% BuildChunk ("fisheye")
 
 %--- calculator --- demo -------------------------------------------------
 % a 10-key calculator using FormsVBT
 % requires: formsvbt
-BuildChunk ("calculator")
+% BuildChunk ("calculator")
 
 %--- cube --- demo -------------------------------------------------
 % a rotating cube
 % requires: formsvbt
-BuildChunk ("cube")
+% BuildChunk ("cube")
 
 %--- codeview --- library -------------------------------------------
 % support for animated views of source code, needed by zeus
 % requires: formsvbt
-BuildChunk ("codeview")
+% BuildChunk ("codeview")
 
 %--- rehearsecode --- utility -------------------------------------------
 % a program to manually test drive source code animations
 % requires: codeview
-BuildChunk ("rehearsecode")
+% BuildChunk ("rehearsecode")
 
 %--- mg --- library -------------------------------------------
 % the low-level animation support
 % requires: formsvbt
-BuildChunk ("mg")
+% BuildChunk ("mg")
 
 %--- mgkit --- library -------------------------------------------
 % a collection of easier-to-use animation widgets
 % requires: m3
-BuildChunk ("mgkit")
+% BuildChunk ("mgkit")
 
 %--- anim3D --- library -------------------------------------------
 % a collection of 3D animation widgets
@@ -480,118 +502,118 @@
 %    X11R4 and PEX
 %    X11R4 and OpenGL
 %    soon: OpenGL and Windows NT
-BuildChunk ("anim3D")
+% BuildChunk ("anim3D")
 
 %--- synloc --- library -------------------------------------------
 % a low-level library used by the Obliq interpreter
 % requires: libm3
-BuildChunk ("synloc")
+% BuildChunk ("synloc")
 
 %--- synex --- library -------------------------------------------
 % a low-level "syntax extension" library used by the Obliq interpreter
 % requires: synloc
-BuildChunk ("synex")
+% BuildChunk ("synex")
 
 %--- metasyn --- library -------------------------------------------
 % a low-level library used by the Obliq interpreter
 % requires: synex
-BuildChunk ("metasyn")
+% BuildChunk ("metasyn")
 
 %--- obliqrt --- library -------------------------------------------
 % the Obliq interpreter's runtime library
 % requires: synloc, netobj
-BuildChunk ("obliqrt")
+% BuildChunk ("obliqrt")
 
 %--- obliqlibm3 --- library -------------------------------------------
 % the Obliq interpreter's hooks to the libm3 library
 % requires: obliqrt, libm3
-BuildChunk ("obliqlibm3")
+% BuildChunk ("obliqlibm3")
 
 %--- obliqlibui --- library -------------------------------------------
 % the Obliq interpreter's hooks to the ui library
 % requires: obliqrt, formsvbt
-BuildChunk ("obliqlibui")
+% BuildChunk ("obliqlibui")
 
 %--- obliqlibanim --- library -------------------------------------------
 % the Obliq interpreter's hooks to the animation libraries
 % requires: obliqrt, mgkit
-BuildChunk ("obliqlibanim")
+% BuildChunk ("obliqlibanim")
 
 %--- obliqparse --- library -------------------------------------------
 % the Obliq interpreter's parser
 % requires: obliqrt
-BuildChunk ("obliqparse")
+% BuildChunk ("obliqparse")
 
 %--- obliqprint --- library -------------------------------------------
 % the Obliq interpreter's value printer
 % requires: obliqrt
-BuildChunk ("obliqprint")
+% BuildChunk ("obliqprint")
 
 %--- obliq --- library -------------------------------------------
 % the Obliq interpreter
 % requires: obliqparse, obliqprint, metasyn
-BuildChunk ("obliq")
+% BuildChunk ("obliq")
 
 %--- obliqlib3D --- library -------------------------------------------
 % the Obliq interpreter's hooks to the 3D animation libraries
 % requires: anim3D, obliqlibanim, obliq
-BuildChunk ("obliqlib3D")
+% BuildChunk ("obliqlib3D")
 
 %--- obliqbinmin --- utility -------------------------------------------
 % the Obliq interpreter with minimal runtime hooks
 % requires: obliq
-BuildChunk ("obliqbinmin")
+% BuildChunk ("obliqbinmin")
 
 %--- obliqbinstd --- utility -------------------------------------------
 % the Obliq interpreter with the "standard" runtime hooks
 % requires: obliq, obliqlibm3
-BuildChunk ("obliqbinstd")
+% BuildChunk ("obliqbinstd")
 
 %--- obliqbinui --- utility -------------------------------------------
 % the Obliq interpreter with ui support
 % requires: obliq, obliqlibm3, obliqlibui
-BuildChunk ("obliqbinui")
+% BuildChunk ("obliqbinui")
 
 %--- obliqbinanim --- utility -------------------------------------------
 % the Obliq interpreter with full animation support
 % requires: obliq, obliqlibm3, obliqlibui, obliqlibanim
-BuildChunk ("obliqbinanim")
+% BuildChunk ("obliqbinanim")
 
 %--- obliqbin3D --- utility -------------------------------------------
 % the Obliq interpreter with full 3D animation support
 % requires: obliq, obliqlibm3, obliqlibui, obliqlibanim, obliqlib3D
-BuildChunk ("obliqbin3D")
+% BuildChunk ("obliqbin3D")
 
 %--- obliqsrvstd --- utility -------------------------------------------
 % an Obliq "server" with the standard runtime hooks
 % requires: obliq, obliqlibm3
-BuildChunk ("obliqsrvstd")
+% BuildChunk ("obliqsrvstd")
 
 %--- obliqsrvui --- utility -------------------------------------------
 % an Obliq "server" with ui support
 % requires: obliq, obliqlibm3, obliqlibui
-BuildChunk ("obliqsrvui")
+% BuildChunk ("obliqsrvui")
 
 %--- m3zume --- utility -------------------------------------------
 % the "interesting event" preprocessor needed by zeus
 % requires: formsvbt, stubgen
-BuildChunk ("m3zume")
+% BuildChunk ("m3zume")
 
 %--- zeus --- library -------------------------------------------
 % the algorithm animation toolkit
 % requires: netobj, codeview, mgkit, m3zume, netobj, stubgen
-BuildChunk ("zeus")
+% BuildChunk ("zeus")
 
 %--- mentor --- demo -------------------------------------------------
 % a collection of algoritm animations
 % requires: zeus, obliqbinanim
-BuildChunk ("mentor")
+% BuildChunk ("mentor")
 
 %--- smalldb --- library -------------------------------------------
 % an in-memory database library, used by the package tools
 % requires: libm3
 if equal (OS_TYPE, "POSIX")
-  BuildChunk ("smalldb")
+%  BuildChunk ("smalldb")
 end
 
 %
@@ -644,37 +666,40 @@
 %--- visualobliq --- demo -------------------------------------------
 % a prototype of an easy-to-use distributed programming environment
 % requires: formsvbt, obliq, obliqlibui, obliqlibm3
-BuildChunk ("visualobliq")
+% BuildChunk ("visualobliq")
 
 %--- voquery --- demo -------------------------------------------
 % a simple query program used by vorun
 % requires: formsvbt
-BuildChunk ("voquery")
+% BuildChunk ("voquery")
 
 %--- vorun --- demo -------------------------------------------
 % a safe visual obliq interpreter suitable for embedding in the WWW.
 % requires: obliqlibui, obliqlibm3, obliq, voquery, vorun
 if equal (OS_TYPE, "POSIX")
-  BuildChunk ("vorun")
+%  BuildChunk ("vorun")
 end
 
 %--- vocgi --- demo -------------------------------------------
 % an HTML/cgi gateway, required to embed Visual Obliq code in the WWW.
 % requires: obliqlibm3, obliq
-BuildChunk ("vocgi")
+% BuildChunk ("vocgi")
 
 %--- llscan --- utility -------------------------------------------
 % a little mh program used by Postcard
 % requires: 
 if equal (OS_TYPE, "POSIX")
-  BuildChunk ("llscan")
+% FreeBSD port: Don't uncomment the following line.  This chunk depends
+% upon absolute pathnames.  It will be built automatically during the
+% installation process.
+%  BuildChunk ("llscan")
 end
 
 %--- postcard --- utility -------------------------------------------
 % an integrated mail/news reader
 % requires: formsvbt llscan mtex
 if equal (OS_TYPE, "POSIX")
-  BuildChunk ("postcard")
+%  BuildChunk ("postcard")
 end
 
 %--- gnuemacs --- library -------------------------------------------------
@@ -682,67 +707,67 @@
 %   also a program to build Modula-3 tags
 % requires: 
 if equal (OS_TYPE, "POSIX")
-  BuildChunk ("gnuemacs")
+%  BuildChunk ("gnuemacs")
 end
 
 %--- webvbt --- library -------------------------------------------------
 % a library for displaying HTML pages inside a VBT
 % requires: formsvbt obliqparse obliqlibm3 obliqlibui obliqlibanim
-BuildChunk ("webvbt")
+% BuildChunk ("webvbt")
 
 %--- webscape --- utility -------------------------------------------------
 % a web browser with support for interactive content
 % requires: webvbt
-BuildChunk ("webscape")
+% BuildChunk ("webscape")
 
 %--- deckscape --- utility -------------------------------------------------
 % a web browser that uses a new metaphor: "decks" of web pages
 % requires: webvbt
-BuildChunk ("deckscape")
+% BuildChunk ("deckscape")
 
 %--- webcard --- utility -------------------------------------------------
 % an integrated mail/news/web client
 % requires: webvbt
 if equal (OS_TYPE, "POSIX")
-  BuildChunk ("webcard")
+%  BuildChunk ("webcard")
 end
 
 %--- ocr --- library -------------------------------------------------
 % interface to optical character recognition library (DECstation only)
 % requires: libm3
-BuildChunk ("ocr")
+% BuildChunk ("ocr")
 
 %--- lecterndoc --- library -------------------------------------------------
 % "lectern" document format
 % requires: libm3
-BuildChunk ("lecterndoc")
+% BuildChunk ("lecterndoc")
 
 %--- lecternclient --- library & utility --------------------------------------
 % tool for sending requests to Lectern server
 % requires: libm3, mtex
-BuildChunk ("lecternclient")
+% BuildChunk ("lecternclient")
 
 %--- lecterntohtml --- utility ------------------------------------------------
 % tool for sending requests to Lectern server
 % requires: lecterndoc, mtex
 if equal (OS_TYPE, "POSIX")
-  BuildChunk ("lecterntohtml")
+%  BuildChunk ("lecterntohtml")
 end
 
 %--- lectern --- utility -------------------------------------------------
 % the "virtual paper" document viewer
 % requires: formsvbt, images, tempfiles, lecternclient
-BuildChunk ("lectern")
+% BuildChunk ("lectern")
 
 %--- buildlectern --- utility -------------------------------------------------
 % program for building lectern documents
 % requires: ocr, lecterndoc, tempfiles, images, mtex
-BuildChunk ("buildlectern")
+% BuildChunk ("buildlectern")
 
 %--- editlectern --- utility -------------------------------------------------
 % program for editing lectern documents
 % requires: formsvbt, lecternclient, lecterndoc
-BuildChunk ("editlectern")
+% BuildChunk ("editlectern")
 
 %--- tcpextras --- library -------------------------------------------------
 % Additions to the tcp library
@@ -752,9 +777,9 @@
 %--- http --- library -------------------------------------------------
 % library for hypertext transfer protocol (HTTP)
 % requires: tcpextras
-BuildChunk ("http")
+% BuildChunk ("http")
 
 %--- webcat --- utility -------------------------------------------------
 % program that takes a URL and prints out the web document
 % requires: http
-BuildChunk ("webcat")
+% BuildChunk ("webcat")
